openmls/src/treesync/mod.rs

```diff
@@ -434,15 +445,52 @@ impl TreeSync {
     /// Returns the nodes in the tree ordered according to the
     /// array-representation of the underlying binary tree.
     pub fn export_nodes(&self) -> Vec<Option<Node>> {
-        self.tree
-            .export_nodes()
-            .iter()
-            // Filter out private keys
-            .map(|node| match node {
-                TreeNode::Leaf(leaf) => leaf.node_without_private_key().map(Node::LeafNode),
-                TreeNode::Parent(parent) => parent.node_without_private_key().map(Node::ParentNode),
-            })
-            .collect()
+        let mut nodes = Vec::new();
+
+        // Get the max length of the tree, blanks beyond the last leaf are
+        // trimmed.
+        let max_length = self.rightmost_full_leaf();
+
+        let mut leaves = self
```

Comment:
Since it took us a while to figure out what the best approach is, it might make sense to add some documentation here.